pkg/grid/src/nav/search/Providers.tsx

```diff
@@ -29,18 +47,25 @@ export const Providers = ({ match, history }: ProvidersProps) => {
     getProviders();
   }, [provider]);
 
-  const handleSearch = useCallback(
-    debounce((input: string) => {
-      push(match?.path.replace(':ship', input.trim()));
-    }, 300),
-    [path]
-  );
-
   useEffect(() => {
-    if (searchInput) {
-      handleSearch(searchInput);
+    if (results) {
+      useLeapStore.setState({
+        matches: results.map((p) => ({ value: p.shipName, display: p.nickname }))
+      });
     }
-  }, [searchInput]);
+  }, [JSON.stringify(results)]);
```

Comment:
yeesh, is there no way to ensure that results isn't recreated every render?

Reply:
this might have been before I had `useMemo` on results. I tried it just now without the stringify and it seems fine.